barman/config.py

```diff
@@ -750,6 +764,78 @@ def __init__(self, config, name):
             if value is not None and value == "" or value == "None":
                 value = None
             setattr(self, key, value)
+        self._active_model_file = os.path.join(
+            self.backup_directory, ".active-model.auto.conf"
```

Comment:
I had assumed we'd store this in the config directory but using `backup_directory` like this seems like a better idea - the active model feels more like state rather than configuration so belongs in the server directory. I wonder if we should remove the `.conf` suffix so that it looks less like a config file (if it looks like a config file users are more likely to want to edit it directly which we don't want happening here).

Reply:
Thanks for your thoughts 🙌 . I think that makes sense, yes. I'll perform the following changes:

* Change the "active" file name
* Move the models stuff out of the `servers -> SERVER -> config` key in the `barman diagnose` output
* Adjust the unit tests, and include the ones which are missing